client/landing/gutenboarding/index.tsx

```diff
@@ -30,7 +32,26 @@ window.AppBoot = () => {
 
 		ReactDom.render(
 			<BrowserRouter basename="gutenboarding">
-				<Gutenboard />
+				<Switch>
+					<Route exact path={ path }>
+						<Gutenboard />
+					</Route>
+					<Route
+						exact
+						path={ `/:step(${ steps.join( '|' ) })/:garbage?` }
+						render={ ( { match } ) => <Redirect to={ makePath( match.params.step ) } /> }
+					/>
+					<Route
+						exact
+						path={ `/:garbage?/:lang(${ langs.join( '|' ) })?` }
+						render={ ( { match } ) => (
+							<Redirect to={ makePath( Step.IntentGathering, match.params.lang ) } />
+						) }
+					/>
```

Comment:
Do we need the above two fallback route defs? I'd really like to get rid of the conditional `garbage` route fragments; I think it's confusing to assign to a named param when we don't need it.

Normally, I think we should be able to write fallback routes without needing named route fragments. What about the route def below e.g.? It's falling back to redirecting to `/` now. That used to be Intent Gathering, but now it isn't. Is this used at all anymore? Can we use it as our fallback? It'd be a bit more easily readable :slightly_smiling_face: 

I think it would help to write down conceptually how we want our routing (and fallbacks to work), e.g.
- If the route matches a step, render that step.
- If it doesn't route a step, redirect to intent gathering.
- ...

Reply:
> It's falling back to redirecting to / now. That used to be Intent Gathering, but now it isn't. Is this used at all anymore?

This is a mistake from a previous struggle to use `usePath()` within AppBoot. There was at one point `makePath()` was removed and `usePath()` could not be used within AppBoot so I resorted to using `/`. See here https://github.com/Automattic/wp-calypso/pull/39312#discussion_r379333583. And it is now using `Step.IntentGathering` again with this commit which I've just commited. https://github.com/Automattic/wp-calypso/pull/39312/commits/72a7258e6df8653e788d58f3c24a20d4e6e66cfb

And as for the `garbage` route fragments, I'm all in for a more elegant solution than this. Haven't got any ideas though, lol.

Reply:
> And as for the `garbage` route fragments, I'm all in for a more elegant solution than this. Haven't got any ideas though, lol.

Let's approach it "top-down" -- first define what kind of behavior we want to see, and based on that, implement it. :slightly_smiling_face:  That's what I meant with

> I think it would help to write down conceptually how we want our routing (and fallbacks to work), e.g.
> - If the route matches a step, render that step.
> - If it doesn't route a step, redirect to intent gathering.
> - ...

OTOH, it'd be good if we can verbalize what each of the current `<Route />`s does, so we can compare, and maybe infer what kind of changes we need to make :slightly_smiling_face:

Reply:
From our zoom call we decided to get rid of garbage route fallback. https://github.com/Automattic/wp-calypso/pull/39312/commits/1188fe67417b125973d3abfd15ae18ba6aee4711